### PR TITLE
`Development`: Unify file endpoints

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [X] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [X] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [X] I documented the Java code using JavaDoc style.

### Motivation and Context

This follows up #5574 and covers all endpoints that use file tokens for authentication.

### Description

I added a new annotation, replaced the occurrences and added the annotation to the documentation.
There are no changes to the mapping as all endpoints in question already have the path prefix `api/files/`

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Upload a file somewhere (e.g. lecture)
2. Try to download the file as a student
3. All should work fine as before

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by Strohgelaender — APPROVED** [review]
code

**Review by dangcosta — APPROVED** [review]
tested on ts1 and worked as expected

File: src/main/java/de/tum/in/www1/artemis/security/annotations/EnforceFile.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.security.access.prepost.PreAuthorize;
+
+/**
+ * This annotation is used to enforce an endpoint using a file token authentication.
+ * It should only be used with endpoints starting with /api/file/
```
Comment (b-fein) [inline src/main/java/de/tum/in/www1/artemis/security/annotations/EnforceFile.java:12]: In `FileResource` the endpoints consistently use `/api/files/`.
```suggestion
 * It should only be used with endpoints starting with {@code /api/files/}.
```

**Review by b-fein — APPROVED** [review]
One small suggestion for the JavaDoc, code looks good otherwise.

**Review by minrows — COMMENTED** [review]
- Tested on ts1, downloading files works as explained.
- Not sure if it's related to this PR, but when I try to load exercise detail page, I get the following error. 
- URL: https://artemis-test1.artemis.in.tum.de/course-management/11/file-upload-exercises/1126

image

File: src/main/java/de/tum/in/www1/artemis/security/annotations/EnforceFile.java (hunk 1 above)
Comment (b-fein) [reply src/main/java/de/tum/in/www1/artemis/security/annotations/EnforceFile.java]: Should the convention be `/api/file/` or `/api/files/`? Here it’s singular, the `FileResource` uses the plural variant.

File: src/main/java/de/tum/in/www1/artemis/security/annotations/EnforceFile.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.security.access.prepost.PreAuthorize;
+
+/**
+ * This annotation is used to enforce an endpoint using a file token authentication.
+ * It should only be used with endpoints starting with {@code /api/file/}
```
Comment (b-fein) [inline src/main/java/de/tum/in/www1/artemis/security/annotations/EnforceFile.java:12]:
```suggestion
 * It should only be used with endpoints starting with {@code /api/files/}.
```
Should the convention be `/api/file/` or `/api/files/`? Here it’s singular, the `FileResource` uses the plural variant.

Comment (julian-christl) [reply src/main/java/de/tum/in/www1/artemis/security/annotations/EnforceFile.java:12]: My bad

**Comment by julian-christl** [comment]
> * Tested on ts1, downloading files works as explained.
> * Not sure if it's related to this PR, but when I try to load exercise detail page, I get the following error.
> * URL: https://artemis-test1.artemis.in.tum.de/course-management/11/file-upload-exercises/1126
> 
> image

Thanks for reporting. This was an issue from #5574. I fixed it in the latest commit there.

**Review by minrows — APPROVED** [review]

**Pull request merged**